collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -260,6 +260,7 @@ func severityTextToNumber(severityText string) plog.SeverityNumber {
 		"FATAL4":   plog.SeverityNumberFatal4,
 		"CRITICAL": plog.SeverityNumberFatal,
 		"ALL":      plog.SeverityNumberTrace,
+		"WARNING":  plog.SeverityNumberWarn,
```

Comment:
Adding `WARNING` here. It is because Python logger can log `WARNING` level when the function record is in JSON format. 
Example
```
{
    "timestamp": "2024-07-12T20:06:53Z",
    "level": "WARNING",
    "message": "30 warning {'filters': [], 'name': 'root', 'level': 0, 'parent': None, 'propagate': True, 'handlers': [<LambdaLoggerHandlerWithFrameType (NOTSET)>], 'disabled': False, '_cache': {10: True, 20: True, 30: True}}",
    "logger": "root",
    "requestId": "dcbdc55b-25bb-49dd-8273-64a96181ca2a",
    "warning": [
        "from-extra",
        30
    ]
}
```